.gitignore

```diff
@@ -1,2 +1,3 @@
 /target
 **/*.rs.bk
+Cargo.lock
```

Comment:
Why did you choose to remove `Cargo.lock`?

https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
I always assumed that .lock files should not be included in git. Isn't it a representation of my locally cached crates?
(I honestly never opened it before you asked)

Reply:
Nope, this is only true for library-type projects. For client-facing applications (such as `sorting-visualization`) build-reproducibility is crucial, so lockfiles are used to pin dependency versions used while development, so that the project can be compiled anywhere/anytime (ideally, but this works good enough). Please, check out the link I provided.